scale/batch/definition/definition.py

```diff
@@ -49,8 +50,19 @@ def validate(self, batch):
                 self.prev_batch_diff.set_force_reprocess(self.forced_nodes)
             if not self.prev_batch_diff.can_be_reprocessed:
                 raise InvalidDefinition('PREV_BATCH_NO_REPROCESS', 'Previous batch cannot be reprocessed')
-        # New batch
         
+        # New batch - need to validate dataset parameters against recipe revision
+        elif self.dataset:
+            from data.models import DataSet
+            from recipe.models import RecipeType
+            dataset = DataSet.objects.get(pk=self.dataset)
+            dataset_definition = dataset.get_definition()
+            recipe_type = RecipeType.objects.get(name=batch.recipe_type.name, revision_num=batch.recipe_type_rev.revision_num)
+            recipe_inputs = recipe_type.get_definition().get_input_keys()
+
+            # No recipe inputs match the dataset 
+            if not any(elem in recipe_inputs for elem in dataset_definition.param_names):
+                raise InvalidDefinition('NO_MATCHING_PARAMS', 'No parameters in the dataset match the recipe type inputs')
```

Comment:
Good, but we need to make sure the dataset has all of the required parameters for the recipe.  You can do this by calling recipe_definition.input_interface.validate_connection(dataset_definition.parameters), although then you'll be missing the global parameters so you'd need to combine the two interface objects and then call the validate_connection method.